include/kframework/state-utils.md

```diff
@@ -183,6 +186,30 @@ The `"rlp"` key loads the block information.
     rule <k> load "rlp"        : (VAL:String => #rlpDecode(#unparseByteStack(#parseByteStack(VAL)))) ... </k>
     rule <k> load "genesisRLP" : (VAL:String => #rlpDecode(#unparseByteStack(#parseByteStack(VAL)))) ... </k>
  // ---------------------------------------------------------------------------------------------------------
+    rule <k> load "rlp" : [ [ HP , HO , HC , HR , HT , HE , HB , HD , HI , HL , HG , HS , HX , HM , HN , HF , WR , .JSONs ] , BT , BU , BW , .JSONs ]
+          => load "transaction" : BT
+          ~> load "withdraw" : BW
+         ...
+         </k>
+         <previousHash>      _ => #asWord(#parseByteStackRaw(HP)) </previousHash>
+         <ommersHash>        _ => #asWord(#parseByteStackRaw(HO)) </ommersHash>
+         <coinbase>          _ => #asWord(#parseByteStackRaw(HC)) </coinbase>
+         <stateRoot>         _ => #asWord(#parseByteStackRaw(HR)) </stateRoot>
+         <transactionsRoot>  _ => #asWord(#parseByteStackRaw(HT)) </transactionsRoot>
+         <receiptsRoot>      _ => #asWord(#parseByteStackRaw(HE)) </receiptsRoot>
+         <logsBloom>         _ => #parseByteStackRaw(HB)          </logsBloom>
+         <difficulty>        _ => #asWord(#parseByteStackRaw(HD)) </difficulty>
+         <number>            _ => #asWord(#parseByteStackRaw(HI)) </number>
+         <gasLimit>          _ => #asWord(#parseByteStackRaw(HL)) </gasLimit>
+         <gasUsed>           _ => #asWord(#parseByteStackRaw(HG)) </gasUsed>
+         <timestamp>         _ => #asWord(#parseByteStackRaw(HS)) </timestamp>
+         <extraData>         _ => #parseByteStackRaw(HX)          </extraData>
+         <mixHash>           _ => #asWord(#parseByteStackRaw(HM)) </mixHash>
+         <blockNonce>        _ => #asWord(#parseByteStackRaw(HN)) </blockNonce>
+         <baseFee>           _ => #asWord(#parseByteStackRaw(HF)) </baseFee>
+         <withdrawalsRoot>   _ => #asWord(#parseByteStackRaw(WR)) </withdrawalsRoot>
+         <ommerBlockHeaders> _ => BU                              </ommerBlockHeaders>
+
     rule <k> load "rlp" : [ [ HP , HO , HC , HR , HT , HE , HB , HD , HI , HL , HG , HS , HX , HM , HN , .JSONs ] , BT , BU , .JSONs ]
           => load "transaction" : BT
          ...
```

Comment:
Is the only difference here between the new rule and the old one the fact that now we also have `HF`, `WR`, and `BW`?

If so, perhaps the new rule could deal with just those new parameters, and then reduce to the old rule? It would be less code duplication I guess. Not sure though.

Reply:
That's a great idea! I removed the code duplication.